### PR TITLE
Attendee flow

### LINKED ISSUE #396: Item assignment UI unintuitive
See the discussion in #387.

### PR DESCRIPTION
**Pull request by melinath**
This is fully functional and pretty awesome. It could use some UI love – making it more obvious when something *can* be selected / *is* selected – which should probably happen as part of this branch.

This will resolve #396 and partially address #403 when it lands. At least that's the idea.

### CODE REVIEW

**Comment by melinath** [comment]
(It also needs some explanatory text, which I would be happy to take a stab at or leave to someone else.)

File: brambling/static/brambling/attendees.ractive.js [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+	function handleKeydown(event) {
+		if (event.which === 32) {
+			// space key
+			this.__tap_handler__.fire();
+		}
+	}
+
+	return ractive_events_tap;
+
+}));
```
Comment (harrislapiroff) [inline brambling/static/brambling/attendees.ractive.js]: Can we put libs in separate files?

Comment (melinath) [inline brambling/static/brambling/attendees.ractive.js]: Sure, I'll do that quick.

**Comment by melinath** [comment]
The ractive shop seems to not be working quite right on this branch. Not sure if it's working on master, but... I'd kinda rather resolve that in a new branch since we're not using it yet anyway? *hopeful / sheepish grin?*

**Comment by harrislapiroff** [comment]
Yeah, that sounds reasonable. I'll keep reviewing this.

**Comment by harrislapiroff** [comment]
It would be preferable to me if this branch still auto-assigned items when only registering a single attendee. That's the most common case and should be as smooth as possible.

File: brambling/templates/brambling/event/order/_attendees_ractive.html [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+					</li>
+				{{/if}}
+			</ul>
+			<div class='panel-footer'>
+				{{#if selectedItem}}
+					<span class='btn btn-default btn-block disabled'>
+						Return to shop for more
+					</span>
+				{{else}}
+					<a class='btn btn-default btn-block' href='../shop'>
```
Comment (harrislapiroff) [inline brambling/templates/brambling/event/order/_attendees_ractive.html]: This URL problem is going to get out of hand. We probably need to write a URL reversing API or something.

Comment (harrislapiroff) [inline brambling/templates/brambling/event/order/_attendees_ractive.html]: Not in this branch.

File: brambling/templates/brambling/event/order/attendee_basic_data.html [hunk 3]
```diff
@@ -147,8 +164,16 @@ <h3 class="margin-leader-0">
 						</div>
 					</div>
 				{% endif %}
-				<button class='btn btn-primary' type="submit">{% if attendee.pk %}Save changes{% else %}Save and continue{% endif %}</button>
-				{% if attendee.pk %}<a href="../" class="btn btn-link">Cancel</a>{% endif %}
+				{% if attendee.pk %}
+					<button class='btn btn-primary' type="submit" name='next' value='review'>Save changes</button>
+					<a href="../" class="btn btn-link">Cancel</a>
```
Comment (harrislapiroff) [inline brambling/templates/brambling/event/order/attendee_basic_data.html]: This is a django template though and should definitely use `{% url %}`

File: brambling/templates/brambling/event/order/attendee_basic_data.html [hunk 4]
```diff
@@ -147,8 +164,16 @@ <h3 class="margin-leader-0">
...
+				{% else %}
+					<button class='btn btn-primary' type="submit" name='next' value='review'>Continue</button>
+					<button class='btn btn-default' type="submit" name='next' value='add'>Register another attendee</button>
+					{% if attendees.count > 0 %}
+						<a href="../" class="btn btn-link">Cancel</a>
```
Comment (harrislapiroff) [inline brambling/templates/brambling/event/order/attendee_basic_data.html]: supra

**Comment by melinath** [comment]
That should be doable. It would also be neat if one could move more than one item at once... Should be doable as long as they're all from the same source

**Comment by harrislapiroff** [comment]
Does the masthead need to be ractive here? We need the countdown timer back on the attendee page.

**Comment by harrislapiroff** [comment]
(Also I don't like making entire pages a single ractive instance, as previously discussed :p)

**Comment by harrislapiroff** [comment]
Clicking the trash cans to delete attendees isn't working.

**Comment by melinath** [comment]
I can fix the trash cans.

I guess the banner image doesn't have to be ractive, but the stepbar does. It has to react to whether the current page is finished or not.

File: brambling/templates/brambling/event/order/attendee_basic_data.html (hunk 4 above)
Comment (melinath) [inline brambling/templates/brambling/event/order/attendee_basic_data.html]: supra?

Comment (harrislapiroff) [inline brambling/templates/brambling/event/order/attendee_basic_data.html]: see above

**Comment by harrislapiroff** [comment]
Hm. I can see that being good, though I don't think it would be a big deal if the stepbar _didn't_ react since it doesn't react on other pages.

In my ideal world, the stepbar would be a completely separate component that would check the attendee component for completion. Especially since it will be reused on multiple pages ultimately. If necessary a parent component could coordinate that communication.

**Comment by melinath** [comment]
yeah, I think having it as a separate component would be really cool.

**Comment by harrislapiroff** [comment]
This looks pretty good actually. Great work on the overlays. Everything _could_ be spiffed up a bit, but I'm happy to get any of that and the other stuff I complained about after merging this. Feelings on merging as-is @melinath?

**Comment by melinath** [comment]
Yep, I'm satisfied with the current state for now. :-)

**Comment by harrislapiroff** [comment]
Let me give it one more pass.

**Comment by harrislapiroff** [comment]
Actually, is it possible to bypass the item assignment page altogether if it's a single-attendee situation?

**Comment by harrislapiroff** [comment]
I can take a swing at that.

**Comment by melinath** [comment]
@harrislapiroff I don't think we should bypass it altogether. It's really easy to accidentally click the "Continue" button from attendee creation instead of "Register another attendee" – and it's not obvious how to go back and add another person if you intended to have more than one.

**Comment by harrislapiroff** [comment]
I think it's very confusing (and a waste of time) to be presented with an interface that you don't have to do anything with and since single-attendee is the most common case, I'd prefer to optimize for that.

**Comment by melinath** [comment]
Okay.

**Comment by harrislapiroff** [comment]
If we continue to get confusion about registering multiple attendees, I'll reconsider. But right now, my instinct says that we'll get more confusion this way.

**Comment by harrislapiroff** [comment]
@melinath Would you review that last commit and then merge if this is good?

**Comment by melinath** [comment]
On it @harrislapiroff

**Pull request merged**